storage/txcache/txCache.go

```diff
@@ -49,6 +53,10 @@ func (cache *TxCache) AddTx(txHash []byte, tx data.TransactionHandler) (ok bool,
 		return
 	}
 
+	if cache.maybeEvictionInProgress.IsSet() {
```

Comment:
I think we should add txs even if eviction is happening - new txs maybe better than those in eviction - or wait here till eviction is finished and add after.

Reply:
I've thought of using a buffered channel to collect the arriving transactions while eviction is in progress, and consuming the channel upon its completion. Though, this could have complicated the matter of `addition handlers` attached to the `shardedTxPool`, which depend on the result of `addTx` (and this result would have been a fake / false `ok, added` result in the implementation using a channel).

Therefore, in order to avoid some undesired effects and not complicate the design at this step (we could have called the `addition handlers` upon the delayed insertion), I've reverted to the previous logic for this matter. Now we do not ignore the transactions that arrive during eviction. We add them in the cache. They do update the score of the senders, which could lead to bugs (like seeing the same sender twice, in two different bins), but we fix this by deciding the eviction order early, and stick to it no matter happens.

Reply:
Even when the cache is full, receiving and putting new elements in it is a must because of the following scenario - it happens many times: validator requests missing transactions when processing the block, the transactions arrive and the interceptor puts them into the cache. In receivedTransaction function (transactionPreProcessor)- which is notified by the txCache, the missing transaction is added into the map used for current block processing. If all the missing transactions were received in the given time, the block will be signed and the consensus continues.

Reply:
I've also added the logic to make a snapshot of the senders early during eviction.